app/services/api/user/active_commodities_service.rb

```diff
@@ -16,62 +17,123 @@ def initialize(subscription)
       def call
         return {} if uploaded_commodity_codes.blank?
 
-        # load the current candidate commodities
-        active_candidates = ::GoodsNomenclature
-          .actual
-          .with_leaf_column
-          .where(goods_nomenclatures__goods_nomenclature_sid: subscription_target_ids)
-          .all
+        {
+          active: active_commodity_codes.sort.uniq,
+          expired: expired_commodity_codes.sort.uniq,
+          invalid: invalid_commodity_codes.sort.uniq,
```

Comment:
I think it might make sense for the methods to have already sorted and removed duplicates. That avoids having to do that here, and e.g. in line 31